src/core/geometry/qgssfcgalgeometry.h

```diff
@@ -0,0 +1,456 @@
+
+    //! Copy constructor
+    QgsSfcgalGeometry( const QgsSfcgalGeometry &otherGeom );
+
+    /**
+     * Returns the underlying SFCGAL geometry
+     * This operation is always fast, as the SFCGAL geometry representation is maintained for the lifetime of the QgsSfcgalGeometry object.
+     */
+    SIP_SKIP sfcgal::shared_geom sfcgalGeometry() const { return mSfcgalGeom; }
+
+    /**
+     * Returns the geometry converted to a QGIS geometry object.
+     * This method is slow to call, as it always involves re-conversion of the underlying SFCGAL geometry object.
+     */
+    std::unique_ptr<QgsAbstractGeometry> asQgisGeometry( QString *errorMsg = nullptr ) const;
+
+    /**
+     * Returns type of the geometry as a WKB type (point / linestring / polygon etc.)
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    Qgis::WkbType wkbType( QString *errorMsg = nullptr ) const;
+
+    /**
+     * Returns type of the geometry as a OGC string in CamelCase
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QString geometryType( QString *errorMsg = nullptr ) const SIP_HOLDGIL;
+
+    /**
+     * Clones the geometry by performing a deep copy
+     */
+    std::unique_ptr<QgsSfcgalGeometry> clone() const;
+
+    /**
+     * Creates a new geometry from a WKB byte pointer
+     * \param wkbPtr WKB byte pointer
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    bool fromWkb( const QgsConstWkbPtr &wkbPtr, QString *errorMsg = nullptr );
+
+    /**
+     * Export the geometry as WKB
+     *
+     * \param flags argument specifies flags controlling WKB export behavior (since QGIS 3.14).
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QByteArray asWkb( QgsAbstractGeometry::WkbFlags flags = QgsAbstractGeometry::WkbFlags(), QString *errorMsg = nullptr ) const;
+
+    /**
+     * Creates a new geometry from a WKT string.
+     * \param wkt WTK string
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    bool fromWkt( const QString &wkt, QString *errorMsg = nullptr );
+
+    /**
+     * Export the geometry as WKT
+     *
+     * \param precision Floating point precision for WKT coordinates. Setting to -1 yields rational number WKT (not decimal) f.e. "Point(1/3, 1/6, 1/4)".
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QString asWkt( int precision = -1, QString *errorMsg = nullptr ) const;
+
+    /**
+     * Returns the closure of the combinatorial boundary of the geometry (ie the topological boundary of the geometry).
+     * For instance, a polygon geometry will have a boundary consisting of the linestrings for each ring in the polygon.
+     * \param errorMsg Error message returned by SFGCAL
+     * \returns boundary for geometry. May be NULLPTR for some geometry types.
+     */
+    std::unique_ptr<QgsSfcgalGeometry> boundary( QString *errorMsg = nullptr ) const SIP_FACTORY;
```

Comment:
If you return unique_ptr then sip conversion already knows that factory is implied:

```suggestion
    std::unique_ptr<QgsSfcgalGeometry> boundary( QString *errorMsg = nullptr ) const;
```